lib/order.rb

```diff
@@ -0,0 +1,54 @@
+require './lib/menu.rb'
+
+class Order
+  attr_reader :choice
+  def initialize(menu = Menu)
+    @menu = menu.new
+    @choice = []
+  end
+
+  def see_menu
+    @menu.list
+  end
+
+  def choose_item(name, count = 1)
+    puts "#{count}x #{name}(s) added to your basket"
+    count.times { @choice << {name => @menu.get_price(name)} }
+  end
+
+  def order_summary
+    self.work_out_order
+  end
+
+  def total_price
+    t = 0
+    @choice.each do |i|
+      i.each_value { |v| t += v }
+    end
+    t
+  end
+
+  def print_receipt
+    # @choice.each do |i|
+    #   i.each {|k, v| puts "#{k} - £%0.2f" % v }
+    # end
+    self.work_out_order
+    puts "Total to pay: £%0.2f" % self.total_price
+  end
+
+  def work_out_order
+    number = {}
+    count = 1
+    @choice.each do |h|
+      h.each_key do |k|
+        if !number.has_key?(k)
```

Comment:
Could this method be simplified? Perhaps the quantity could always be stored (even if it's one) to negate the need of the conditional here?